Rename FindCommand to SearchCommand

johannagwan opened the pull request:
Resolves #24 
 
**main:**
- Rename FindCommand.java to SearchCommand.java (in src/main/java/seedu/address/logic/commands/SearchCommand.java)
- Rename FindCommandParser.java to SearchCommandParser.java (in src/main/java/seedu/address/logic/parser/SearchCommandParser.java)
- Update AddressBookParser.java accordingly (in src/main/java/seedu/address/logic/parser/AddressBookParser.java)

**test:**
- Rename FindCommandTest.java to SearchCommandTest.java (in src/test/java/seedu/address/logic/commands/SearchCommandTest.java)
- Update AddressBookParserTest.java accordingly (in src/test/java/seedu/address/logic/parser/AddressBookParserTest.java)
- Rename FindCommandParserTest.java to SearchCommandParserTest.java (in src/test/java/seedu/address/logic/parser/SearchCommandParserTest.java)

kevinputera commented:
@johannagwan I haven't taken a look at this, will do soon.

Nonetheless, do keep a lookout at the Travis CI checks for every PR you submit. If it fails, that generally means: (1) there are compilation errors, (2) there are stylistic errors, or (3) there are test errors.

Failing checks will most likely happen quite frequently for this project since we're refactoring heavily; as such, most of the time we might need to fix the test cases to accommodate for the refactoring. However, I believe that this PR should not break, since it only involves renaming, so do fix this!

kevinputera commented:
Another thing, let's add the `status.onGoing` tag to issue #24 so that we can track our progress easier in the milestone + let others know that you are working on this issue. In addition, let's [link](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) this PR to the issue, so that the issue is automatically closed when this PR is merged.

I have done the above, but writing this here for future reference

johannagwan commented:
Hi @kevinputera @HemanshuGandhi, thanks for the review! Just want to clarify, I did not add any inline comments at all. But it's a good catch! We all can take note of this in the future.

I have amended my PR as such:
- added .vs directory into gitignore
- removed unnecessary commits (I'm sorry for the mishaps)

kevinputera commented on src/main/java/com/notably/logic/commands/AddCommand.java:
```diff
@@ -63,4 +64,4 @@ public boolean equals(Object other) {
                 && toAdd.equals(((AddCommand) other).toAdd));
     }
-}
+}
```
You need a newline character after the end of the file. Otherwise, this will account to a style error (although we can't really check right now since the build is failing)

kevinputera replied:
You need a newline character after the end of the file. Otherwise, this will account to a style error (although we can't really check right now since the build is failing)

kevinputera commented on src/main/java/com/notably/logic/commands/EditCommand.java:
```diff
@@ -19,6 +19,11 @@
+import com.notably.model.person.Email;
+import com.notably.model.person.Name;
+import com.notably.model.person.Person;
+import com.notably.model.person.Phone;
```
Please remove all the above imports. These conflicted with the removal of these imports from #48 (merged PR)

kevinputera commented on src/main/java/com/notably/model/Model.java:
```diff
@@ -4,6 +4,7 @@
 import java.util.function.Predicate;
 
 import com.notably.commons.core.GuiSettings;
+import com.notably.model.person.Person;
```
Please remove this import, conflict with #48

kevinputera commented on src/test/java/com/notably/logic/commands/DeleteCommandTest.java:
```diff
@@ -11,19 +11,12 @@
-import seedu.address.model.ModelManager;
-import seedu.address.model.UserPrefs;
->>>>>>> main/master:src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
+import com.notably.model.person.Person;
```
Please remove this import, conflict with #48

kevinputera commented on src/test/java/com/notably/logic/parser/AddCommandParserTest.java:
```diff
@@ -1,136 +1,6 @@
-        assertParseFailure(parser, PREAMBLE_NON_EMPTY + NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
-    }
```
How do these removals relate to this PR?

kevinputera commented on src/test/java/com/notably/logic/parser/AddressBookParserTest.java:
```diff
@@ -19,11 +19,12 @@
 import com.notably.logic.commands.ListCommand;
 import com.notably.logic.parser.exceptions.ParseException;
 import com.notably.model.person.NameContainsKeywordsPredicate;
+import com.notably.model.person.Person;
```
Please remove this, conflict with #48

kevinputera commented on src/test/java/com/notably/testutil/EditPersonDescriptorBuilder.java:
```diff
@@ -5,6 +5,11 @@
+import com.notably.model.person.Email;
+import com.notably.model.person.Name;
+import com.notably.model.person.Person;
+import com.notably.model.person.Phone;
```
Please remove these imports, conflict with #48

kevinputera commented on src/test/java/com/notably/testutil/PersonBuilder.java:
```diff
@@ -3,6 +3,11 @@
+import com.notably.model.person.Email;
+import com.notably.model.person.Name;
+import com.notably.model.person.Person;
+import com.notably.model.person.Phone;
```
Same here too, conflict with #48

kevinputera commented on src/test/java/com/notably/testutil/PersonUtil.java:
```diff
@@ -10,6 +10,7 @@
 
 import com.notably.logic.commands.AddCommand;
 import com.notably.logic.commands.EditCommand.EditPersonDescriptor;
+import com.notably.model.person.Person;
```
Same here, conflict with #48

johannagwan commented:
I am closing this PR as we have decided to integrate the Search feature into our SuggestionEngine. I will create a new PR for the deletion of FindCommand.
